expression/builtin_miscellaneous_vec.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"encoding/binary"
+	"math"
+	"net"
+
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+func (b *builtinInetNtoaSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ReserveString(n)
+	i64s := buf.Int64s()
+	for i := 0; i < n; i++ {
+		val := i64s[i]
+		if buf.IsNull(i) || val < 0 || uint64(val) > math.MaxUint32 {
+			result.AppendNull()
+			continue
+		}
+		ip := make(net.IP, net.IPv4len)
```

Comment:
Could we reuse this buffer during for loop?

Reply:
update. PTAL

Reply:
Any benchmark result update?

Reply:
Update. But it seems that the promotion is not obvious.

Reply:
We can use option `benchmem` to see the benchmark of memory.
```
go test -benchmem
```